src/domain/event/hooks/useEventServerErrors.tsx

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { ApolloError, ServerError } from '@apollo/client';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { LEServerError, ServerErrorItem } from '../../../types';
+import lowerCaseFirstLetter from '../../../utils/lowerCaseFirstLetter';
+import pascalCase from '../../../utils/pascalCase';
+
+type ShowServerErrorsFnParams = {
+  callbackFn?: () => void;
+  error: any;
+  eventType: string;
+};
+
+type UseEventServerErrorsState = {
+  serverErrorItems: ServerErrorItem[];
+  setServerErrorItems: (items: ServerErrorItem[]) => void;
+  showServerErrors: (params: ShowServerErrorsFnParams) => void;
+};
+
+const useEventServerErrors = (): UseEventServerErrorsState => {
```

Comment:
This is much better now but I'm still wondering if most of these functions could be left out of the hook. Meaning that `parseEventServerErrors` could be the helper function defined outside of this hook.

The issue now I feel is that there is a lot of functions to scroll before seeing the important reused hook logic `showServerErrors`. Also if `parseEventServerErrors` is moved outside, it would be nice to have 

```
    return Array.isArray(result)
      ? result.reduce(..
```

in the top of the function so it could be read from top to bottom, not from bottom (in that case function statements should be used to leverage hoistin)

Reply:
Moved parseEventServerErrors to separate file and flipped order of helper function inside it to make it more readable